src/Workspaces/Core/Portable/FindSymbols/FindReferences/Finders/LinkedFileReferenceFinder.cs

```diff
@@ -22,6 +19,15 @@ internal class LinkedFileReferenceFinder : IReferenceFinder
             {
                 var originalDocument = solution.GetDocument(location.SyntaxTree);
 
+                // GetDocument will return null for locations in #load'ed trees.
+                // TODO:  Remove this check and add logic to fetch the #load'ed tree's
+                // Document once https://github.com/dotnet/roslyn/issues/5260 is fixed.
+                if (originalDocument == null)
+                {
+                    Debug.Assert(solution.Workspace.Kind == "Interactive");
```

Comment:
No, it should not be assumed that all SyntaxTree's have documents.  You should always guard against null and have an exit strategy.